keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,213 @@
+---
+kep-number: draft-20180426
+title: Security Conformance
+authors:
+  - "@easeway"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-testing
+reviewers:
+  - "@tallclair"
+  - "@ericchiang"
+  - "@liggitt"
+  - "@davidopp"
+approvers:
+  - "@tallclair"
+  - "@ericchiang"
+  - "@liggitt"
+editor: "@easeway"
+creation-date: 2018-04-26
+status: provisional
+---
+
+# Security Conformance
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Conformance Profiles](#conformance-profiles)
+    * [Behavioral Criteria](#behavioral-criteria)
+    * [Non-behavioral Criteria](#non-behavioral-criteria)
+    * [Implementation Details](#implementation-details)
+* [Graduation Criteria](#graduation-criteria)
+
+## Summary
+
+Security Conformance defines how a Kubernetes cluster will be considered as secure,
+and also proposes the mechanisms to configure a cluster securely.
+
+## Motivation
+
+Configuring a Kubernetes cluster is complicated,
+and it's very easy to mis-configure a cluster to be insecure.
+The current Kubernetes conformance test suite barely covers security aspects.
+
+### Goals
+
+- A list of behavioral security criteria that can be validated by tests
+- A list of non-behavioral security criteria that can be validated by tests
+
+### Non-Goals
+
+- Mechanisms to satisfy security criteria
+- Details about Kubernetes conformance test
+
+## Proposal
+
+### Conformance Profiles
+
+The behavioral/non-behavioral criteria are categorized into a few conformance profiles for different levels of security requirements:
+
+- `Essential`: the baseline security requirements for the cluster,
+focusing on the threaten from outside the cluster (via APIs),
+non-privileged (non-root) access on the host running master and kubelet,
+not covering the threaten from user workloads (non-critical, non-system Pods) running inside the cluster.
+The criteria in this category must always be satisfied;
+- `Restricted`: in addition to `Essential`,
+it covers the threaten from user workloads to the cluster,
+not among user workloads (e.g. single-tenancy model);
+- `Isolated`: in addition to `Restricted`, it covers the threaten among user workloads (e.g. multi-tenancy model).
+
+### Behavioral Criteria
+
+#### Config File Permissions
+
+  *Conformance Profile: Essential*
+
+  The configuration files for API server, controller manager, scheduler and kubelet should have the correct setup of ownership and file permissions.
+
+  - non-root user should not be able to access etcd data directory
+  - non-root user should not be able to access any kubeconfig files
+  - non-root user should not be able to access any private key files
+
+#### Anonymous Access
+
+  *Conformance Profile: Essential*
+
+  Anonymous access to Kubernetes API should be denied as unauthorized.
```

Comment:
I think denying anonymous access is unreasonable. I will remove this. Because at least for healthz checks, it should allow anonymous access, from external monitoring systems. For discovery, swagger/openapi, it's OK to allow anonymous access. For the rest of APIs, as long as authorizers will block "system:unauthorized", it won't be a security concern.